x-pack/plugins/ml/public/application/timeseriesexplorer/components/series_controls/series_controls.tsx

```diff
@@ -0,0 +1,290 @@
+    });
+  }, 500);
+
+  const entityFieldValueChanged: EntityControlProps['entityFieldValueChanged'] = (
+    entity,
+    fieldValue
+  ) => {
+    const resultEntities = {
+      ...entityControls.reduce((appStateEntities, appStateEntity) => {
+        appStateEntities[appStateEntity.fieldName] = appStateEntity.fieldValue;
+        return appStateEntities;
+      }, {} as Record<string, any>),
+      [entity.fieldName]: fieldValue,
+    };
+
+    appStateHandler(APP_STATE_ACTION.SET_ENTITIES, resultEntities);
+  };
+
+  const detectorIndexChangeHandler: EuiSelectProps['onChange'] = useCallback(
+    (e) => {
+      const id = e.target.value;
+      if (id !== undefined) {
+        appStateHandler(APP_STATE_ACTION.SET_DETECTOR_INDEX, +id);
+      }
+    },
+    [appStateHandler]
+  );
+
+  const detectorSelectOptions = detectors.map((d) => ({
+    value: d.index,
+    text: d.detector_description,
+  }));
+
+  const onFieldConfigChange: EntityControlProps['onConfigChange'] = useCallback(
+    (fieldType, config) => {
+      const updatedFieldConfig = {
+        ...(resultFieldsConfig[fieldType] ? resultFieldsConfig[fieldType] : {}),
+        ...config,
+      } as UiPartitionFieldConfig;
+
+      const updatedResultConfig = { ...resultFieldsConfig };
+
+      if (resultFieldsConfig[fieldType]?.anomalousOnly !== updatedFieldConfig.anomalousOnly) {
+        // In case anomalous selector has been changed
+        // we need to change it for all the other fields
+        for (const c in updatedResultConfig) {
+          if (updatedResultConfig.hasOwnProperty(c)) {
+            updatedResultConfig[c as EntityFieldType]!.anomalousOnly =
+              updatedFieldConfig.anomalousOnly;
+          }
+        }
+      }
+
+      setStorageFieldsConfig({
+        ...updatedResultConfig,
+        [fieldType]: updatedFieldConfig,
+      });
+    },
+    [resultFieldsConfig, setStorageFieldsConfig]
+  );
+
+  /** Indicates if any of the previous controls is empty */
+  let hasEmptyFieldValues = false;
+
+  return (
+    <div data-test-subj="mlSingleMetricViewerSeriesControls">
+      <EuiFlexGroup>
+        <EuiFlexItem grow={false}>
+          <EuiFormRow
+            label={
+              <FormattedMessage
+                id="xpack.ml.timeSeriesExplorer.detectorLabel"
+                defaultMessage="Detector"
+              />
+            }
+          >
+            <EuiSelect
+              onChange={detectorIndexChangeHandler}
+              value={selectedDetectorIndex}
+              options={detectorSelectOptions}
+              data-test-subj="mlSingleMetricViewerDetectorSelect"
+            />
+          </EuiFormRow>
+        </EuiFlexItem>
+        {entityControls.map((entity) => {
+          const entityKey = `${entity.fieldName}`;
+          const forceSelection = !hasEmptyFieldValues && entity.fieldValue === null;
+          hasEmptyFieldValues = !hasEmptyFieldValues && forceSelection;
+          return (
+            <EntityControl
```

Comment:
Wondering if it's worth adding an `EuiHealth` style indicator next to each value to indicate its max anomaly score?

![image](https://user-images.githubusercontent.com/7405507/97452233-6cd3b000-192c-11eb-96bc-5ad82f93e9f7.png)

I think it would need to indicate the max record score for the time range in the view.

Reply:
Added in https://github.com/elastic/kibana/pull/81510/commits/af2f0efe0d3e7bd257b39c002031e06826d97641